lib/elf/file.rb

```diff
@@ -374,12 +374,16 @@ def is_compatible(other)
       raise TypeError.new("wrong argument type #{other.class} (expected Elf::File)") unless
         other.is_a? Elf::File
 
+      compatible_abi = case
+        when [@abi, other.abi].include?(Elf::OsAbi::SysV); true
```

Comment:
@Flameeyes you can find updated commit with `gnu_specific.c` that results in ELF file that have OS ABI `GNU/Linux`. `gcc` tries to produce `SysV` always.
Also from `binutils` sources I found that `ifunc` supported by both `FreeBSD` and `GNU/Linux` OS ABI's. From `glibc` it looks like it threats `SysV` and `GNU/Linux` almost equally.